include/eve/detail/function/simd/riscv/bit_cast.hpp

```diff
@@ -0,0 +1,569 @@
+//==================================================================================================
+/*
+  EVE - Expressive Vector Engine
+  Copyright : EVE Project Contributors
+  SPDX-License-Identifier: BSL-1.0
+*/
+//==================================================================================================
+#pragma once
+
+#include <eve/as.hpp>
+#include <eve/concept/simd.hpp>
+#include <eve/detail/category.hpp>
+#include <eve/detail/is_native.hpp>
+#include <eve/traits/as_integer.hpp>
+
+namespace eve::detail
+{
+template<plain_scalar_value T, typename N>
+EVE_FORCEINLINE wide<T, typename N::combined_type>
+                riscv_lmul_ext(wide<T, N> a) noexcept
+requires rvv_abi<abi_t<T, N>>
+{
+  constexpr auto c = categorize<wide<T, typename N::combined_type>>();
+  static_assert(N::combined_type::value != N::value);
+  constexpr auto in_lmul  = riscv_rvv_dyn_::getLMUL<T>(N::value);
+  constexpr auto out_lmul = riscv_rvv_dyn_::getLMUL<T>(N::combined_type::value);
+  if constexpr( out_lmul == in_lmul ) return a.storage();
+  if constexpr( match(c, category::float64) )
+  {
+    if constexpr( out_lmul == 2 ) return __riscv_vlmul_ext_f64m2(a);
+    if constexpr( out_lmul == 4 ) return __riscv_vlmul_ext_f64m4(a);
+    if constexpr( out_lmul == 8 ) return __riscv_vlmul_ext_f64m8(a);
+  }
```

Comment:
we do else after return in if constexpr chains.

Reply:
Yep, I thought about it, but for RISC-V this will made a huge chain of `if constexpr (...) ... else if constexpr (...) ... else if constexpr (...) ...` that will make code less readable and understandable. I try to return each if statement, so while debugging problems it will be easier to find problem

Reply:
We have those huge chains, this is how we do it

Reply:
Done